### PR TITLE
12.0 website sale fix fja

### PR DESCRIPTION
**Pull request by fja-odoo**
Now a product removed from wishlist as a connected user can be added again.

In the product page the found count was not display

It was possible to add the product itself as alternative product
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale_wishlist/models/product_wishlist.py [hunk 1]
```diff
@@ -37,7 +37,19 @@ def current(self):
 
     @api.model
     def _add_to_wishlist(self, pricelist_id, currency_id, website_id, price, product_id, partner_id=False):
-        wish = self.env['product.wishlist'].create({
+        if partner_id:
+            wish = self.search([('partner_id', '=', partner_id), ('product_id', '=', product_id), '|', ('active', '=', False), ('active', '=', True)])
```
Comment (qsm-odoo) [inline addons/website_sale_wishlist/models/product_wishlist.py]:
```python
self.with_context(active_test=False).search([('partner_id', '=', partner_id), ('product_id', '=', product_id)])
```

File: addons/website_sale_wishlist/models/product_wishlist.py [hunk 2]
```diff
@@ -37,7 +37,19 @@ def current(self):
...
+            if wish:
+                wish.update({
+                    'active': True,
+                    'currency_id': currency_id,
+                    'pricelist_id': pricelist_id,
+                    'price': price,
+                    'website_id': website_id,
```
Comment (qsm-odoo) [inline addons/website_sale_wishlist/models/product_wishlist.py]: I would create a dict with all the values and use the same dict in both update and create. Also why `update` and not `write` ?

File: addons/website_sale/controllers/main.py [hunk 3]
```diff
@@ -315,12 +315,16 @@ def product(self, product, category='', search='', **kwargs):
...
+        domain = self._get_search_domain(search, category, attrib_values)
+        product_count = request.env['product.template'].search_count(domain)
+
         if not product_context.get('pricelist'):
             product_context['pricelist'] = pricelist.id
             product = product.with_context(product_context)
 
         values = {
             'search': search,
+            'search_count': product_count,
```
Comment (qsm-odoo) [inline addons/website_sale/controllers/main.py]: Where is this used ? "fix fond in product" ?

File: addons/website_sale/views/templates.xml [hunk 4]
```diff
@@ -1565,7 +1565,8 @@
                 <td colspan="3" class="text-center text-xl-right noborder">
                 <span class=''>
                     <t t-set='force_coupon' t-value="website_sale_order.pricelist_id.code or request.params.get('code_not_available')"/>
-                    <t t-if="not force_coupon">
+                    <t t-set='sale_coupon_active' t-value="request.env['ir.module.module'].sudo().search([('name', '=', 'sale_coupon'), ('state', '=', 'installed')])"/>
+                    <t t-if="not force_coupon and sale_coupon_active">
```
Comment (qsm-odoo) [inline addons/website_sale/views/templates.xml]: - Please reread your commit messages before sending to review...
- ... and no it is not ok to do a search to check an external module is installed ... that part should be moved to sale_coupon but if it crashes in stable, I'd say the correct fix is to check that code is in the fields.

File: addons/website_sale/controllers/main.py (hunk 3 above)
Comment (fja-odoo) [reply addons/website_sale/controllers/main.py]: It is the found count in product page. When you perform a research with the "Show # found" customization enabled. I Will update the commit message accordingly.

File: addons/website_sale/views/templates.xml (hunk 4 above)
Comment (fja-odoo) [reply addons/website_sale/views/templates.xml]: Yes I got kind of confused in this...

File: addons/website_sale/controllers/main.py (hunk 3 above)
Comment (qsm-odoo) [reply addons/website_sale/controllers/main.py]: JKE decided as I explained to you: make the xml change to not show the number of found on a product page

File: addons/website_sale/views/templates.xml (hunk 4 above)
Comment (fja-odoo) [reply addons/website_sale/views/templates.xml]: The code is part of website sale but is used in coupon only...

Comment (qsm-odoo) [reply addons/website_sale/views/templates.xml]: @rdeodoo

Comment (rdeodoo) [reply addons/website_sale/views/templates.xml]: discussed with fja, pricelist can act as a coupon code when the pricelist has an ecommerce promotional code

**Comment by qsm-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**